dependencydiff/raw_dependencies.go

```diff
@@ -0,0 +1,180 @@
+	if err != nil {
+		return nil, fmt.Errorf("error creating the github repo: %w", err)
+	}
+
+	// Initialize the checks to run from the caller's input.
+	checksToRun := checker.CheckNameToFnMap{}
+	if checkNamesToRun == nil && len(checkNamesToRun) == 0 {
+		// If no check names are provided, we run the default checks for the caller.
+		for _, cn := range defaultChecksToRun {
+			checksToRun[cn] = checks.AllChecks[cn]
+			// Use the check name to decide if we need to create the three special clients below.
+		}
+	} else {
+		for _, cn := range checkNamesToRun {
+			checksToRun[cn] = checks.AllChecks[cn]
+		}
+	}
+
+	// Initialize the client(s) corresponding to the checks to run.
+	ghRepoClient := githubrepo.CreateGithubRepoClient(ctx, logger)
+	// Initialize these three clients as nil at first.
+	var ossFuzzClient clients.RepoClient = nil
+	var vulnsClient clients.VulnerabilitiesClient = nil
+	var ciiClient clients.CIIBestPracticesClient = nil
+	// Create the corresponding client if the check needs to run.
+	for cn := range checksToRun {
+		switch cn {
+		case checks.CheckFuzzing:
+			ossFuzzClient, err = githubrepo.CreateOssFuzzRepoClient(ctx, logger)
+			if err != nil {
+				return nil, err
+			}
+		case checks.CheckVulnerabilities:
+			vulnsClient = clients.DefaultVulnerabilitiesClient()
+		case checks.CheckCIIBestPractices:
+			ciiClient = clients.DefaultCIIBestPracticesClient()
+		}
+	}
+
+	results := []pkg.DependencyCheckResult{}
+	for _, d := range deps {
+		// Skip removed dependencies and don't run scorecard checks on them.
+		if *d.ChangeType == pkg.Removed {
+			continue
+		}
+		depCheckResult := pkg.DependencyCheckResult{
+			PackageURL:       d.PackageURL,
+			SourceRepository: d.SourceRepository,
+			ChangeType:       d.ChangeType,
+			ManifestPath:     d.ManifestPath,
+			Ecosystem:        d.Ecosystem,
+			Version:          d.Version,
+			Name:             d.Name,
+		}
+		// For now we skip those without source repo urls.
+		// TODO: use the BigQuery dataset to supplement null source repo URLs
+		// so that we can fetch the Scorecard results for them.
+		if d.SourceRepository != nil && *d.SourceRepository != "" {
+			// If the srcRepo is valid, run scorecard on this dependency and fetch the result.
+			// TODO: use the Scorecare REST API to retrieve the Scorecard result statelessly.
+			scorecardResult, err := pkg.RunScorecards(
+				ctx,
+				ghRepo,
+				// TODO: In future versions, ideally, this should be
+				// the commitSHA corresponding to d.Version instead of HEAD.
+				clients.HeadSHA,
+				checksToRun,
+				ghRepoClient,
+				ossFuzzClient,
+				ciiClient,
+				vulnsClient,
+			)
+			// "err==nil" suggests the run succeeds, so that we record the scorecard check results for this dependency.
+			// Otherwise, it indicates the run fails and we leave the current dependency scorecard result empty
+			// rather than letting the entire API return nil since we still expect results for other dependencies.
+			if err == nil {
```

Comment:
where do you set the error if scorecard returns an error?

Reply:
can i record the scorecard failing errors into the logger?